models/discordactions.js

```diff
@@ -114,32 +116,56 @@ const updateDiscordImageForVerification = async (userDiscordId) => {
   }
 };
 
-const getNumberOfMemberForGroups = async (groups = []) => {
+/**
+ * Enriches group data with membership information for a given Discord ID.
+ *
+ * @param {string} discordId - The Discord ID of the user.
+ * @param {Array<object>} groups - Array of group objects to process.
+ * @returns {Promise<Array<object>>} - An array of group objects with enriched information.
+ */
+const enrichGroupDataWithMembershipInfo = async (discordId, groups = []) => {
   try {
     if (!groups.length) {
       return [];
     }
-    const roleIds = groups.map((group) => group.roleid);
 
+    const groupCreatorIds = groups.reduce((ids, group) => {
+      if (group.createdBy) {
+        ids.add(group.createdBy);
+      }
+      return ids;
+    }, new Set());
+
+    const groupCreatorsDetails = await retrieveUsers({ userIds: Array.from(groupCreatorIds) });
```

Comment:
Nit: Hey @bksh05, Only you are using retrieveUsers function call in model 
I seen whole backend file user retrieveUsers function call in controller
please discuss with others also about this point??

Reply:
retrieveUsers is a function used to fetch users. It should be fine wherever we use it. 
We should use retrieveUsers whenever we want to fetch user data because it removes sensitive info.

Reply:
Resolving this in favour of the previous comment